docs/flow/modin/data_management/functions.rst

```diff
@@ -3,41 +3,104 @@ Function Module Description
-* GroupBy functions — Evaluates GroupBy aggregation for that type of functions that can be executed via MapReduce approach. To be able to form groups engine efficiently broadcasts `by` partitions to each partition of the source frame.
-* Map functions — Apply function to each partition in parallel. Note, that the map function should not change the shape of the partitions.
-* MapReduce functions — Function that reduces specified axis into a scalar. First applies map function to each partition in parallel, then concatenates resulted partitions along the specified axis and applies reduction function. Note that the execution engine expects that the reduction function returns a scalar.
-* Reduction functions — Function that reduces specified axis into a scalar, but requires knowledge about the whole axis. Be aware that providing this knowledge may be expensive because the execution engine has to concatenate partitions along the specified axis. Also, note that the execution engine expects that the reduction function returns a scalar.
-* Default-to-pandas functions — Do fallback to pandas for passed function.
+Map functions
+-------------
+Apply function to each partition in parallel. **Note**: map function should not change the shape of the partitions.
 
-Each template represented with a class with the corresponding name and implements ``register`` method, which takes functions to apply in an appropriate way and instantiate the related template. Functions that are passed to the ``register`` will be executed under deserialized and preprocessed (depends on the template) partitions, so the function would take one of the pandas object: ``pandas.DataFrame``, ``pandas.Series`` or ``pandas.DataFrameGroupbyObject``.
+.. figure:: /img/map_evaluation.svg
+    :align: center
+
+Reduction functions
+-------------------
+Function that reduces specified axis into a scalar, but requires knowledge about the whole axis.
+Be aware that providing this knowledge may be expensive because the execution engine has to
+concatenate partitions along the specified axis. Also, note that the execution engine expects
+that the reduction function returns a one dimensional frame.
+
+.. figure:: /img/reduce_evaluation.svg
+    :align: center
+
+MapReduce Functions
+-------------------
+Function that reduces specified axis into a scalar. First applies map function to each partition
+in parallel, then concatenates resulted partitions along the specified axis and applies reduction
+function. In contrast with `Map function` template, here you're allowed to change partition shape
+in the map phase. Note that the execution engine expects that the reduction function returns a one dimensional frame.
+
+Binary functions
+----------------
+Function that takes two operands (left is always `QueryCompiler`) and evaluates function along them.
+Engine efficiently broadcasts partitions of the right operand to the left if necessary.
```

Comment:
did you mean "effectively"?

Reply:
I meant "efficiently", but now I realized that we shouldn't grade our implementations in docs, so removed that confusing word.